commit e5f9df1ac59925e12588c6edf4a371175090d203
Author: Keno Fischer <keno@juliacomputing.com>
Date:   Sat Jun 22 19:42:28 2019 -0400

    [Support] Fix build under Emscripten
    
    Summary:
    Emscripten's libc doesn't define MNT_LOCAL, thus causing a build
    failure in the fallback path. However, to the best of my knowledge,
    it also doesn't support remote file system mounts, so we may simply
    return `true` here (as we do for e.g. Fuchsia). With this fix, the
    core LLVM libraries build correctly under emscripten (though some
    of the tools and utils do not).
    
    Reviewers: kripken
    
    Subscribers: kristina, llvm-commits
    
    Tags: #llvm
    
    Differential Revision: https://reviews.llvm.org/D63688

diff --git a/lib/Support/Unix/Path.inc b/lib/Support/Unix/Path.inc
index 2ecb97316c8..ea4c8b9db47 100644
--- a/lib/Support/Unix/Path.inc
+++ b/lib/Support/Unix/Path.inc
@@ -380,6 +380,9 @@ static bool is_local_impl(struct STATVFS &Vfs) {
 #elif defined(__CYGWIN__)
   // Cygwin doesn't expose this information; would need to use Win32 API.
   return false;
+#elif defined(__EMSCRIPTEN__)
+  // Emscripten doesn't currently support remote filesystem mounts.
+  return true;
 #elif defined(__sun)
   // statvfs::f_basetype contains a null-terminated FSType name of the mounted target
   StringRef fstype(Vfs.f_basetype);
